### PR TITLE
exp/client/horizon: use protocols/horizon

### LINKED ISSUE #996: exp/clients/horizon: Move new response structs to protocols/horizon
Move new responses from `responses.go` to `protocols/horizon` and use structs directly from this package.

### PR DESCRIPTION
**Pull request by poliha**
This PR adds new response structs to `protocols/horizon`
Also the new horizon client starts using it for responses.
Closes #996

### CODE REVIEW

File: protocols/horizon/main.go [hunk 1]
```diff
@@ -475,3 +477,134 @@ func MustKeyTypeFromAddress(address string) string {
 
 	return ret
 }
+
```
Comment (ire-and-curses) [inline protocols/horizon/main.go]: Docstring

File: protocols/horizon/main.go [hunk 2]
```diff
@@ -475,3 +477,134 @@ func MustKeyTypeFromAddress(address string) string {
...
+	RequestsTotal          LogTotalMetric `json:"requests.total"`
+	CoreLatestLedger       SingleMetric   `json:"stellar_core.latest_ledger"`
+	CoreOpenConnections    SingleMetric   `json:"stellar_core.open_connections"`
+	TxsubBuffered          SingleMetric   `json:"txsub.buffered"`
+	TxsubFailed            LogMetric      `json:"txsub.failed"`
+	TxsubOpen              SingleMetric   `json:"txsub.open"`
+	TxsubSucceeded         LogMetric      `json:"txsub.succeeded"`
+	TxsubTotal             LogTotalMetric `json:"txsub.total"`
+}
+
```
Comment (ire-and-curses) [inline protocols/horizon/main.go]: Docstring

File: protocols/horizon/main.go [hunk 3]
```diff
@@ -475,3 +477,134 @@ func MustKeyTypeFromAddress(address string) string {
...
+type Problem struct {
+	Type     string                     `json:"type"`
+	Title    string                     `json:"title"`
+	Status   int                        `json:"status"`
+	Detail   string                     `json:"detail,omitempty"`
+	Instance string                     `json:"instance,omitempty"`
+	Extras   map[string]json.RawMessage `json:"extras,omitempty"`
+}
+
+// ToProblem converts the Prolem to a problem.P
```
Comment (ire-and-curses) [inline protocols/horizon/main.go]: Is there a way to make this description clearer? I'm not sure what a problem.P is or why I need it

File: protocols/horizon/main.go [hunk 4]
```diff
@@ -475,3 +477,137 @@ func MustKeyTypeFromAddress(address string) string {
...
+
+// Problem represents an error response from horizon
+type Problem struct {
+	Type     string                     `json:"type"`
+	Title    string                     `json:"title"`
+	Status   int                        `json:"status"`
+	Detail   string                     `json:"detail,omitempty"`
+	Instance string                     `json:"instance,omitempty"`
+	Extras   map[string]json.RawMessage `json:"extras,omitempty"`
+}
```
Comment (bartekn) [inline protocols/horizon/main.go]: I think we can stick to `support/render/problem.Problem` so it isn't needed here.

Comment (poliha) [reply protocols/horizon/main.go]: okay makes sense. They have similar structs.

**Pull request merged**